File: 2015/src/test/java/nl/sbmf21/aoc15/days/Day10Test.java
package nl.sbmf21.aoc15.days;

import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.HashMap;

import static nl.sbmf21.aoc.testing.UtilKt.buildWithInput;
import static nl.sbmf21.aoc.testing.UtilKt.testDay;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class Day10Test {

    @Test
    public void testInput() {
        testDay(Day10.class, 492982, 6989950, false, null);
    }

    @Test
    public void testExample() {
        Day10 day = buildWithInput(Day10.class, new ArrayList<>());

        new HashMap<String, String>() {{
            put("1", "11");
            put("11", "21");
            put("21", "1211");
            put("1211", "111221");
            put("111221", "312211");
        }}.forEach((line, expected) -> assertEquals(expected, day.lookAndSay(line)));
    }
}

File: 2021/kotlin/src/test/kotlin/nl/sbmf21/aoc21/days/Day10Test.kt
package nl.sbmf21.aoc21.days

import nl.sbmf21.aoc.testing.testDay
import kotlin.test.Test

class Day10Test {

    @Test
    fun testInput() = testDay(Day10::class.java, 343_863, 2_924_734_236L)

    @Test
    fun testExample() = testDay(Day10::class.java, 26_397, 288_957L, true)
}